text/0000-compiler-major-change-process.md

```diff
@@ -0,0 +1,255 @@
+
+Here are some examples of changes that were made in the past that would warrant the major change process:
+
+* overhauling the way we encode crate metadata
+* merging the gcx, tcx arenas
+* renaming a widely used, core abstraction, such as the `Ty` type
+* introducing cargo pipelining 
+* adding a new `-C` flag that exposes some minor variant
+
+## What are some examples of things that are too big for the major change process?
+
+Here are some examples of changes that are too big for the major change process, or which at least would require auxiliary design meetings or a more fleshed out design before they can proceed:
+
+* introducing incremental or the query system
+* introducing MIR or some new IR
+* introducing parallel execution
+* adding ThinLTO support
+
+## What are some examples of things that are too small for the major change process?
+
+Here are some examples of things that don't merit any MCP:
+
+* adding new information into metadata
+* fixing an ICE or tweaking diagnostics
+* renaming "less widely used" methods
+
+## When should Major Change Proposals be closed?
+
+Major Change Proposals can be closed:
+
+* by the author, if they have lost interest in pursuing it.
+* by a team lead or expert, if there are strong objections from key
+  members of the team that don't look likely to be overcome.
+* by folks doing triage, if there have been three months of
+  inactivity. In this case, people should feel free to re-open the
+  issue if they would like to "rejuvenate" it.
+
+# Template for major change proposals
+
+[major change template]: #Template-for-major-change-proposals
+
+The template for major change proposals is as follows:
+
+```
+# What is this issue?
+
+This is a **major change proposal**, which means a proposal to make a notable change to the compiler -- one that either alters the architecture of some component, affects a lot of people, or makes a small but noticeable public change (e.g., adding a compiler flag). You can read more about the MCP process on https://forge.rust-lang.org/.
+
+**This issue is not meant to be used for technical discussion. There is a Zulip stream for that. Use this issue to leave procedural comments, such as volunteering to review, indicating that you second the proposal (or third, etc), or raising a concern that you would like to be addressed.**
+
+# MCP Checklist
+
+* [x] Fill out and file this issue. The @rust-lang/wg-prioritization group will add this to the triage meeting agenda so folks see it.
+* [ ] Create a Zulip topic in the stream `#t-compiler/major changes` with the name `XXX compiler-team#NNN`, where `XXX` is the title of this issue and `NNN` is whatever issue number gets assigned. Discuss the MCP in there, and feel free to update the proposal as needed.
+* [ ] Find a **reviewer**, and add their name to this comment (see the section below).
+* [ ] Find a **second**, someone who is knowledgeable of the area and approves of the design, and have them leave a comment on the issue.
```

Comment:
This requirement was also unclear. @pnkfelix clarified (in comment https://github.com/rust-lang/compiler-team/issues/278#issuecomment-620216953) that the "second" should come from the compiler team.

Can this be changed to:

* [ ] Find a **second** from the compiler team, ...

or

* [ ] Find a **second**, preferably someone from the compiler team who is knowledgeable...

Reply:
I think saying a "compiler team member or contributor" is probably right.